app/controllers/posts_controller.rb

```diff
@@ -1,31 +1,61 @@
 class PostsController < ApplicationController
+  load_and_authorize_resource
+
   def index
     @posts = Post.all
     @user = User.includes(posts: [:comments]).find(params[:user_id].to_i)
   end
 
+  def current_ability
+    @current_ability ||= Ability.new(current_admin)
+  end
+
   def new
+    @id = current_admin.id
     @post = Post.new
+    @id = current_admin.id
   end
 
   def create
-    @post = Post.new(post_params)
-    @author = current_user
-    @post.author = @author
-    if @post.save
-      flash[:success] = 'Post was successfully created'
-      redirect_to new_user_post_path(@author.id)
+    post = Post.new(title: post_params[:title], text: post_params[:text], author: current_admin)
+    @author = post.author
```

Comment:
You can simply create a new post using `current_user` which is automatically handled by `Devise`.  `current_user` automatically becomes the currently logged-in user. Therefore you do not need to worry about setting the `author_id`.
```
   @post = current_user.posts.new(post_params)
```

- `post_params` is a method you already have on your posts controller which expects a post from the page.


And that's it, that's all you have to do. Cancancan and Devise will handle everything else.

Reply:
This will mean restaring the devise process . The project requirement for the devise model creating gave an option to choose between a user model and an admin model.